newrelic/structures_newrelic_alert_channel.go

```diff
@@ -203,12 +202,42 @@ func flattenAlertChannelConfiguration(c *alerts.ChannelConfiguration, d *schema.
 	configResult["recipients"] = c.Recipients
 	configResult["region"] = c.Region
 	configResult["route_key"] = c.RouteKey
-	configResult["service_key"] = c.ServiceKey
 	configResult["tags"] = c.Tags
 	configResult["teams"] = c.Teams
-	configResult["url"] = c.URL
 	configResult["user_id"] = c.UserID
 
+	if attr, ok := d.GetOk("config.0.api_key"); ok {
```

Comment:
There are 12 sensitive schema values in the config.  Is this going to cover all cases?

Reply:
I was hoping you'd chime in here. I don't know to be honest, but I also don't want to make too drastic of a change and cause cascading bugs somewhere else. I also [asked additional config examples](https://github.com/newrelic/terraform-provider-newrelic/issues/420#issuecomment-666807402) from some folks in hopes of getting more scenarios covered.

I'll continue to investigate

Reply:
So far we're covered for fields the API considers sensitive, which are `api_key`, `auth_password`, `auth_type`, `route_key`, and `service_key` (I did a diff based on a full POST channel config compared to the GET response config I received). 

We still have some extreme outlier scenarios for `headers`/`headers_string` and `payload`/`payload_string`, but it seems to mostly involve detecting import state and conditionally handling that. I think we can address that in another PR if we deem this worthy.

Bottom line here is that this PR makes things a good bit better than they were regarding the config drift 😎